core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -563,11 +572,17 @@ private synchronized void processTaskUpdate(TaskInfo newValue, List<SplitAssignm
             if (assignment.isNoMoreSplits()) {
                 noMoreSplits.put(planNodeId, false);
             }
+            else {
+                hasMoreSplits = true;
+            }
             if (isPartitionedSource) {
                 pendingSourceSplitCount -= removed;
                 pendingSourceSplitsWeight -= removedWeight;
             }
         }
+        if (pendingSourceSplitsWeight == 0 && hasMoreSplits) {
```

Comment:
shouldn't we use `getQueuedPartitionedSplitsWeight` here?
`pendingSourceSplitCount` only counts splits not yet acknowledged by the worker if I understand correctly.
Also, it only counts split for partitioned sources, what happens for not partitioned ones?